packages/storage_client/lib/src/storage_file_api.dart

```diff
@@ -91,6 +97,106 @@ class StorageFileApi {
     return (response as Map)['Key'] as String;
   }
 
+  /// Upload a file with a token generated from `createUploadSignedUrl`.
+  ///
+  /// [path] The file path, including the file name. Should be of the format `folder/subfolder/filename.png`. The bucket must already exist before attempting to upload.
+  ///
+  /// [token] The token generated from `createUploadSignedUrl`
+  ///
+  /// [file] The body of the file to be stored in the bucket.
+  Future<String> uploadToSignedUrl(
+    String path,
+    String token,
+    File file, [
+    FileOptions fileOptions = const FileOptions(),
+    int? retryAttempts,
+    StorageRetryController? retryController,
+  ]) async {
+    assert(retryAttempts == null || retryAttempts >= 0,
+        'retryAttempts has to be greater or equal to 0');
+
+    final cleanPath = _removeEmptyFolders(path);
+    final _path = _getFinalPath(cleanPath);
+    var url = Uri.parse('${this.url}/object/upload/sign/$_path');
+    url = url.replace(queryParameters: {'token': token});
+
+    await storageFetch.putFile(
+      url.toString(),
+      file,
+      fileOptions,
+      retryAttempts: retryAttempts ?? _retryAttempts,
+      retryController: retryController,
+    );
+
+    return cleanPath;
```

Comment:
So far we are using `(response as Map)['Key'] as String` to return the path in `upload()`, but I see that storage-js uses `cleanPath` everywhere. I guess they are always the same, so it doesn't matter.